### PR TITLE
175 i need a rsibonus module in the aprcalculator contract to handle the calculations of the rsibonus

### PR DESCRIPTION
**Pull request by SamBorisov**

### CODE REVIEW

File: contracts/APRCalculator/modules/RSI/RSIndex.sol [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    function _setRSI(uint256 rsindex) private {
+        uint256 newRsi;
+        if (rsindex > 39) {
+            newRsi = 0;
+        } else if (rsindex > 29) {
+            newRsi = 11500;
+        } else if (rsindex > 19) {
+            newRsi = 12500;
+        } else if (rsindex < 20) {
```
Comment (R-Santev) [inline contracts/APRCalculator/modules/RSI/RSIndex.sol:108]: else is enough here

File: test/APRCalculator/MacroFactor.test.ts [hunk 2]
```diff
@@ -79,12 +69,12 @@ export function RunMacroFactorTests(): void {
...
       const currentMacroFactor = await aprCalculator.getMacroFactor();
       expect(currentMacroFactor).to.equal(INITIAL_DEFAULT_MACRO_FACTOR);
-      expect(await aprCalculator.disabledMacro()).to.be.true;
+      expect(await aprCalculator.disabledBonusesUpdates()).to.be.true;
       expect(await aprCalculator.smaSlowSum()).to.equal(oldSMASlowSum);
       expect(await aprCalculator.smaFastSum()).to.equal(oldSMAFastSum);
     });
 
-    it("should disable gard and enable macro factor updates after calling the function again", async function () {
+    it("should disable guard and enable macro factor updates after calling the function again", async function () {
```
Comment (R-Santev) [inline test/APRCalculator/MacroFactor.test.ts:77]: Is there a test about what would happen if a price update is made when the functionality is disabled. What will be the price when the functionality is re-enabled?

File: test/APRCalculator/RSIndex.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      for (let i = 0; i !== 3; i++) {
+        await aprCalculator.connect(this.signers.system).quotePrice(currentPrice);
+        await commitEpoch(systemHydraChain, hydraStaking, [this.signers.validators[1]], this.epochSize, DAY);
+      }
+
+      expect(await aprCalculator.rsi()).to.be.equal(MAX_RSI_BONUS);
+    });
+  });
+
+  describe("Gard RSIndex", function () {
```
Comment (R-Santev) [inline test/APRCalculator/RSIndex.test.ts:92]:
```suggestion
  describe("Guard RSIndex", function () {
```

**Review by R-Santev — CHANGES_REQUESTED** [review]

File: test/APRCalculator/MacroFactor.test.ts (hunk 2 above)
Comment (SamBorisov) [reply test/APRCalculator/MacroFactor.test.ts:77]: There were tests for the event to determine if it emits or not. Now I added tests to check when we guard and update prices, that the avrageGain/Loss is not changed and when we disable the guard it does it is changed with the newest only update prices that were triggered after the guard is disabled. Also, added tests to check we calculate the RSI correctly!

**Review by R-Santev — APPROVED** [review]

**Pull request merged**